crates/bevy_render/src/render_resource/shader.rs

```diff
@@ -143,19 +144,30 @@ impl ProcessedShader {
                 },
             )?,
         };
-        let module_info = naga::valid::Validator::new(
-            naga::valid::ValidationFlags::default(),
-            naga::valid::Capabilities::default(),
-        )
-        .validate(&module)?;
+        let mut capabilities = naga::valid::Capabilities::empty();
+        if features.contains(Features::PUSH_CONSTANTS) {
```

Comment:
This seems like something we'll need to maintain a larger mapping of eventually. Could we factor this out into a const `[(wgpu::Features, naga::valid::Capabilities); N]` array and just have a for loop run over them to make this easier to expand in the future? Rust *should* unroll the loop if performance is somehow a concern here.

Ideally this should fail to compile if naga ever adds any additional `Capabilities` that aren't covered here, but I'm not sure if that's possible.

Reply:
I'm gonna try and come up with a nice solution for this. Sadly it's neither possible to match bitflags nor to iterate over all valid flags, but there are some workarounds.